README.md

```diff
@@ -35,6 +35,21 @@ fn process_data() -> Result<()> {
 fn unpack_config(data: &str) -> &str {
     "/some/path/that/does/not/exist"
 }
+
+fn main() {
+    if let Err(e) = process_data() {
+        print_error(&e);
+    }
+}
+
+fn print_error(e: &dyn std::error::Error) {
+    eprintln!("error: {}", e);
+    let mut cause = e.source();
+    while let Some(e) = cause {
+        eprintln!("caused by: {}", e);
+        cause = e.source();
+    }
+}
```

Comment:
This is a good start, but I'd rather avoid having people need to copy-paste this code in all of their projects. instead, see [this comment](https://github.com/shepmaster/snafu/issues/110#issuecomment-509245702) where we discuss adding an iterator that does the core part of this function. Then, this would look like

```rust
fn main() {
    if let Err(e) = process_data() {
        eprintln!("error: {}", e);
        for cause in whatever_iterator_is_called(e) {
            eprintln!("caused by: {}", cause);
        }
    }
}
```

Reply:
If I remember correctly, there was a cargo package that kinda does this. Don't recall the specific name. It used the rust [Termination trait]( `https://doc.rust-lang.org/std/process/trait.Termination.html)` and wrapped other Error types with an iterator around it. 

Of course, I like the idea of snafu providing an iterator directly somehow, but just thought it was a nice way to kick start it off for quick programs.. seen quite a few times, where people didn't use it correctly for simple programs, and made me do a face palm :)

Reply:
> there was a cargo package

There's an unstable feature from the standard library ([`iter_chain`](https://doc.rust-lang.org/nightly/std/error/trait.Error.html#method.iter_chain) / [`iter_sources`](https://doc.rust-lang.org/nightly/std/error/trait.Error.html#method.iter_sources)) that basically I'd like to copy into SNAFU for now, add to the `ErrorCompat` trait, then replace with the standard library's version when it stabilizes.

Would you be interested in adding that support to improve this documentation?

Reply:
Sounds good. I'm not sure if I'll be able to make time this week though.. Just opening up an issue for that leaving this PR edited and currently blocked on it. Please feel free to close it if you feel it'd rather be tackled as one unit otherwise :)